int32 pinNumber
float32 degree